homeassistant/components/http.py

```diff
@@ -474,7 +479,8 @@ def handle_request(self, request, **values):
         if self.hass.wsgi.api_password is None:
             authenticated = True
 
-        elif request.remote_addr in self.hass.wsgi.approved_ips:
+        elif any(ip_address(request.remote_addr) in trusted_network
+                 for trusted_network in self.hass.wsgi.trusted_networks):
             authenticated = True
```

Comment:
Even though I'm returning the same result as when password is empty it still asks for a passwort, which I can skip by entering nothing at all.

Where would I need to introduce changes to skip the password dialog completely?